### DISCUSSION TITLE
Replace inner `Vec` in `VecModel`

### DISCUSSION THREAD

**Discussion by rdrpenguin04** [slint-ui/slint#1043] [Ideas]
In my application, fairly often, I don't just need to remove an item from a list and add another one; I need to replace the list with an entirely separate list. I could use `set_<model name>` instead, but then I also have to reset every other function that had the `ModelRc` cloned... needless to say, this is complicated, and it seems like there should be a better way.

It seems like it wouldn't be too hard to have a function to replace the `Vec` in a `VecModel` with an entirely new `Vec`; is there anything that would be blocking this? If not, I could implement it probably.

Edit: It would also be useful to read the inner `Vec` for other purposes, like finding an element with a certain ID.

**Comment by tronical** [comment]
I think that’s a good idea and feature to have. I think this also ties with the ability of the model to notify the view that it’s entire content has changed in one go (the “model reset”).

Would be great if you can help implement this :) and of course we’re happy to assist. Feel also free to change this discussion into an issue (or let me know if you don’t see that option, then I can also do it).

**Reply by rdrpenguin04** [reply]
Alright, I can do that :smile:

**Reply by tronical** [reply]
Regarding the inner vec: maybe a slice accessor could be added (non-mutable). I wonder what others think about that.

edit / side-note: the iter() provides you with an iterator over all model entries, which you can also use to access.

**Reply by rdrpenguin04** [reply]
Alright, `.iter()` solves that problem for me; thank you!
